docker-compose.yaml

```diff
@@ -179,6 +181,65 @@ services:
     volumes:
       - ./cdr/cdm/jupyter/user_shared_workspace:/cdm_shared_workspace/user_shared_workspace
 
+  dev_jupyterhub:
+      build:
+        context: .
+        dockerfile: Dockerfile
+      container_name: dev-jupyterhub
+      ports:
+        - "4043:4043"
+      depends_on:
+        - spark-master
+        - minio-create-bucket
+      environment:
+        - NOTEBOOK_PORT=4043
+        - JUPYTER_MODE=jupyterhub
+        - YARN_RESOURCE_MANAGER_URL=http://yarn-resourcemanager:8032
+        - SPARK_MASTER_URL=spark://spark-master:7077
+        - SPARK_DRIVER_HOST=dev-jupterhub
+        - MINIO_URL=http://minio:9002
+        - MINIO_ACCESS_KEY=minio-readwrite
+        - MINIO_SECRET_KEY=minio123
+        - S3_YARN_BUCKET=yarn
+        - MAX_EXECUTORS=4
+        - POSTGRES_USER=hive
+        - POSTGRES_PASSWORD=hivepassword
+        - POSTGRES_DB=hive
+        - POSTGRES_URL=postgres:5432
+        - USAGE_MODE=dev
+      volumes:
+        - ./cdr/cdm/jupyter:/cdm_shared_workspace
+        - ./cdr/cdm/jupyter/home:/home
+
+  user_jupyterhub:
+    build:
+      context: .
+      dockerfile: Dockerfile
+    container_name: user-jupyterhub
+    ports:
+      - "4044:4044"
+    depends_on:
+      - spark-master
+      - minio-create-bucket
+    environment:
+      - NOTEBOOK_PORT=4044
+      - JUPYTER_MODE=jupyterhub
+      - YARN_RESOURCE_MANAGER_URL=http://yarn-resourcemanager:8032
+      - SPARK_MASTER_URL=spark://spark-master:7077
+      - SPARK_DRIVER_HOST=user-jupyterhub
+      - MINIO_URL=http://minio:9002
+      - MINIO_ACCESS_KEY=minio-readonly
+      - MINIO_SECRET_KEY=minio123
+      - S3_YARN_BUCKET=yarn
+      - JUPYTER_MODE=jupyterhub
+      - MAX_EXECUTORS=4
+      - POSTGRES_USER=hive
+      - POSTGRES_PASSWORD=hivepassword
+      - POSTGRES_DB=hive
+      - POSTGRES_URL=postgres:5432
+    volumes:
+      - ./cdr/cdm/jupyter/home:/home
```

Comment:
No shared workspace?

Reply:
no. Each user will have their own dir and python environment.

Reply:
Should `dev_jupyterhub` have the shared workspace mounted in that case?

Reply:
both instances mount `- ./cdr/cdm/jupyter/jupyterhub/users_home:/jupyterhub/users_home`. Each user will have their own dir inside of `/jupyterhub/users_home` and create notebook there.

Reply:
My point is that `dev-jupyterhub` *also* mounts `/cdr/cdm/jupyter:/cdm_shared_workspace`. I'm asking if that line needs to be there or if it should be removed to match `user_juypterhub`

Reply:
It should be there, at least for now. There are postgres files there. Dev environment should have access to them.

Reply:
I thought we were running a separate postgres server? What are the local files?

Reply:
there is a `hive_metastore` and `cdm-postgres` dir.

Reply:
I don't see any issues with using `- ./cdr/cdm/jupyter:/cdm_shared_workspace` for the dev environment.

Reply:
I'm not saying there's an issue, I just wasn't sure why it was there. can you remind me what's in the hive_metastore and cdm-postgres dirs? Are they both still relevant with the standalone postgres service?

Reply:
I think cdm-postgres is not longer used, it's not getting updated for 2 months. 
In `hive_metastore` there are <db_name>.db files. I think they are used when users run spark select statement. 

![Screenshot 2024-09-03 at 3 01 27 PM](https://github.com/user-attachments/assets/d9a396e3-d1a5-48c3-9b3b-8918be1d0808)

Reply:
Ok, maybe cdm-postgres was the local DB, and hive_metastore is cache data?

Reply:
not exactly sure. But I want to worry about those later.